testsuite/MDAnalysisTests/analysis/test_diffusionmap.py

```diff
@@ -94,3 +96,10 @@ def test_not_universe_error(u):
     trj_only = u.trajectory
     with pytest.raises(ValueError, match='U is not a Universe'):
         diffusionmap.DiffusionMap(trj_only)
+
+
+def test_DistanceMatrix_attr_warning(u):
+    dist = diffusionmap.DistanceMatrix(u, select='backbone').run(step=3)
+    wmsg = f"The `dist_matrix` attribute was deprecated in MDAnalysis 2.0.0"
+    with pytest.warns(DeprecationWarning, match=wmsg):
+        assert_equal(getattr(dist, "dist_matrix"), dist.results.dist_matrix)
```

Comment:
instead of checking values, can't you just check for identity of objects
```python
assert dist.dist_matrix is dist.results.dist_matrix
```
?

Reply:
Admittedlt, we might not have been particularly consistent in testing. Ultimately, this is fine & we could come back and unify if needed.